framework/log/FileTarget.php

```diff
@@ -117,31 +122,21 @@ public function export()
             clearstatcache();
         }
         if ($this->enableRotation && @filesize($this->logFile) > $this->maxFileSize * 1024) {
-            @flock($fp, LOCK_UN);
-            @fclose($fp);
             $this->rotateFiles();
-            $writeResult = @file_put_contents($this->logFile, $text, FILE_APPEND | LOCK_EX);
```

Comment:
@bizley I have mixed feelings here as I've removed something that I don't understand. Why not use `fwrite()` in both cases? I assume there were some problems caused by the messed up locking that someone tried to fix this way.

Reply:
Ok, it dates actually back to the early days of yii2:

https://github.com/yiisoft/yii2/blame/2646e2cdcc6a260c30c8ed24928e2c9a4754d3f5/framework/log/FileTarget.php

There it did not use `ftruncate()` and so the actual logfile did no longer exist after rotation. This logic could have been removed when we introduced `ftruncate()` to clear the original logfile. It should be safe to remove it now.